miio/integrations/vacuum/roborock/vacuumcontainers.py

```diff
@@ -353,6 +534,58 @@ def complete(self) -> bool:
         return self.data["complete"] == 1
 
 
+class FloorCleanDetails(DeviceStatus):
+    """Contains details about a last cleaning run per floor."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+        for map_id in self.data:
+            if self.data[map_id] is None:
+                setattr(self, f"CleanDetails_{map_id}", None)
+                setattr(self, f"start_{map_id}", None)
+                continue
+            setattr(self, f"CleanDetails_{map_id}", self.data[map_id])
+            setattr(self, f"start_{map_id}", self.data[map_id].start)
```

Comment:
This looks rather messy and hard to follow through the setattr "magic", I think we can need to find a better solution for doing this. See my comment above wrt multiroom support.